x-pack/plugins/ml/public/application/datavisualizer/data_comparison/index_patterns_picker.tsx

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC } from 'react';
+import { EuiPageBody, EuiPageSection } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { parse } from 'query-string';
+import { MlPageHeader } from '../../components/page_header';
+import { DataDriftIndexPatternsEditor } from './data_drift_index_patterns_editor';
+
+export const DataDriftIndexPatternsPicker: FC = () => {
+  const { sourceIp, destIp } = parse(location.search, {
+    sort: false,
+  }) as { sourceIp: string; destIp: string };
+
+  const initialProductionIndexPattern = destIp ? destIp.replaceAll(`'`, '') : '';
+  const initialReferenceIndexPattern = sourceIp ? sourceIp.replaceAll(`'`, '') : '';
+
+  return (
+    <div data-test-subj="mlPageSourceSelection">
+      <EuiPageBody restrictWidth={1200}>
+        <MlPageHeader>
+          <FormattedMessage
+            id="xpack.ml.newJob.wizard.selectIndexPatterns"
+            defaultMessage="Specify index patterns"
+          />
+        </MlPageHeader>
+        <EuiPageSection>
+          <DataDriftIndexPatternsEditor
```

Comment:
I've added an option to analyze data drift without saving here, which will create adhoc data views instead of a normal data view. This is visible only if:
1) User can edit/create data views
2) An exisiting data view has not already existed.

<img width="1271" alt="Screen Shot 2023-08-29 at 16 56 57" src="https://github.com/elastic/kibana/assets/43350163/218c37e2-6327-40a0-b61a-82c524d5c9c6">